src/payment/strategies/quadpay/quadpay-payment-strategy.spec.ts

```diff
@@ -0,0 +1,262 @@
+        submitPaymentAction = of(createAction(PaymentActionType.SubmitPaymentRequested));
+
+        jest.spyOn(store, 'dispatch');
+
+        jest.spyOn(paymentMethodActionCreator, 'loadPaymentMethod')
+            .mockResolvedValue(store.getState());
+
+        jest.spyOn(store.getState().checkout, 'getCheckoutOrThrow')
+            .mockReturnValue(checkoutMock);
+
+        jest.spyOn(storeCreditActionCreator, 'applyStoreCredit')
+            .mockReturnValue(applyStoreCreditAction);
+
+        jest.spyOn(remoteCheckoutActionCreator, 'initializePayment')
+            .mockResolvedValue(store.getState());
+
+        jest.spyOn(orderActionCreator, 'submitOrder')
+            .mockReturnValue(submitOrderAction);
+
+        jest.spyOn(requestSender, 'post')
+            .mockReturnValue(Promise.resolve());
+
+        jest.spyOn(paymentActionCreator, 'submitPayment')
+            .mockReturnValue(submitPaymentAction);
+
+        strategy = new QuadpayPaymentStrategy(
+            store,
+            orderActionCreator,
+            paymentActionCreator,
+            paymentMethodActionCreator,
+            storeCreditActionCreator,
+            remoteCheckoutActionCreator,
+            requestSender
+        );
+    });
+
+    describe('#initialize()', () => {
+        it('initializes the strategy successfully', async () => {
+            await expect(strategy.initialize()).resolves.toEqual(store.getState());
+        });
+    });
+
+    describe('#execute()', () => {
+        let orderRequestBody: OrderRequestBody;
+        let quadpayOptions: PaymentInitializeOptions;
+
+        beforeEach(async () => {
+            orderRequestBody = {
+                ...getOrderRequestBody(),
+                payment: {
+                    methodId: 'quadpay',
+                },
+            };
+            quadpayOptions = { methodId: 'quadpay' };
+
+            await strategy.initialize();
+        });
+
+        it('executes the strategy successfully and submits the payment', async () => {
+            const { payment, ...order } = orderRequestBody;
+            const expectedPayment = {
+                methodId: 'quadpay',
+                paymentData: {
+                    nonce: 'checkout_id',
+                },
+            };
+
+            await strategy.execute(orderRequestBody, quadpayOptions);
+
+            expect(storeCreditActionCreator.applyStoreCredit).toHaveBeenCalledWith(false);
+            expect(remoteCheckoutActionCreator.initializePayment).toHaveBeenCalledWith(expectedPayment.methodId, { useStoreCredit: false });
+            expect(orderActionCreator.submitOrder).toHaveBeenCalledWith(order, quadpayOptions);
+            expect(requestSender.post).toHaveBeenCalled();
+            expect(paymentActionCreator.submitPayment).toHaveBeenCalledWith(expectedPayment);
+        });
+
+        it('executes the strategy successfully and applies the store credit', async () => {
```

Comment:
This test and the previous share constants, maybe you can add them on a `describe`  block